lib/active_model/serializer.rb

```diff
@@ -164,6 +164,14 @@ def json_key
       end
     end
 
+    def id
+      object.id if object
+    end
+
+    def type
+      object.class.to_s.demodulize.underscore.pluralize
```

Comment:
I'd prefer raising an error when the `id` attribute is missing since in some cases the `id` used to build the URL isn't the resource's original `id` but a slug maybe.